packages/flutter/lib/src/widgets/form.dart

```diff
@@ -279,7 +283,7 @@ class FormFieldState<T> extends State<FormField<T>> {
 
   /// Calls the [FormField]'s onSaved method with the current value.
   void save() {
-    if (widget.onSaved != null)
+    if (widget.onSaved != null && widget.enabled)
```

Comment:
we should definitely document that this does nothing if enabled is false, but is that even what we want? If someone calls save() on a disabled field, should that work? If not, should it assert?

Reply:
My thinking was that if you disable an input, you don't really care about it anymore, so it should not be saved either. I might be wrong, but it seemed to make sense from the consumer point of view. As for doing an assert, is it really the responsibility of the caller to make such assertions? Should the caller really care about the state of the controls instead of the controls making sure they don't do anything they are not expected to?